client/merchant-feedback-prompt/index.tsx

```diff
@@ -0,0 +1,223 @@
+ *
+ * This is a temporary solution until Gutenberg Snackbar component accepts two actions and WooPayments can render dual-action snackbars using `createNotice`.
+ */
+const WCFooterPortal = ( { children }: { children: React.ReactNode } ) => {
+	const portalRoot = document.getElementsByClassName(
+		'woocommerce-layout__footer'
+	)[ 0 ];
+
+	if ( ! portalRoot ) {
+		return null;
+	}
+
+	return ReactDOM.createPortal( children, portalRoot );
+};
+
+interface MerchantFeedbackPromptProps {
+	/** A function to be called when the user dismisses the prompt and it is to be removed. */
+	dismissPrompt: () => void;
+}
+
+/**
+ * Renders the merchant feedback prompt (snackbar) in the WC footer.
+ *
+ * This is used to gather feedback from merchants about their experience with WooPayments.
+ * Only renders if there are no core notices and the prompt has not been dismissed.
+ *
+ * Note that this includes a customised React component for the snackbar,
+ * because the Snackbar component doesn't accept two actions. See comment below for more details.
+ */
+const MerchantFeedbackPrompt: React.FC< MerchantFeedbackPromptProps > = ( {
+	dismissPrompt,
+} ) => {
+	// Get the core notices, which we'll use to ensure we're not rendering the prompt if there are other notices being displayed.
+	const coreNotices = useSelect(
+		( select ) =>
+			select( 'core/notices' ).getNotices() as NoticeList.Notice[]
+	);
+
+	// Create a ref to track if the view event has been recorded to prevent multiple recordings on a single screen.
+	const hasRecordedViewEvent = useRef( false );
+
+	// Only render the prompt if there are no core notices.
+	const shouldShowPrompt = coreNotices?.length === 0;
+
+	useEffect( () => {
+		// Record the event when the prompt is rendered, but only once per screen.
+		if ( shouldShowPrompt && ! hasRecordedViewEvent.current ) {
+			recordEvent( 'wcpay_merchant_feedback_prompt_view' );
+			hasRecordedViewEvent.current = true;
+		}
+	}, [ shouldShowPrompt ] );
+
+	if ( ! shouldShowPrompt ) {
+		return null;
+	}
+
+	return (
+		<WCFooterPortal>
+			<SnackbarList
+				className="wcpay-merchant-feedback-prompt-wrap"
+				notices={ [
+					{
+						id: 'wcpay-merchant-feedback-prompt',
+						className: 'wcpay-merchant-feedback-prompt',
+						/**
+						 * The custom content for the snackbar is required because the Snackbar / Snackbar component doesn't accept two actions.
+						 * Once this is resolved in Gutenberg and available for WooPayments, we can remove this custom React content,
+						 * and provide the actions to the `actions` prop, using `createNotice` to add the snackbar without requiring a custom React component or portal.
+						 *
+						 * See https://github.com/WordPress/gutenberg/blob/c300edfebb48f79f6f0f6643ce04dd73303c5fcb/packages/components/src/snackbar/index.tsx#L119-L126
+						 */
+						content: (
+							<Flex
```

Comment:
Using `VStack` could be a simpler way to achieve the same layout

Reply:
Thanks @ciampo, TIL about `VStack`!

Unfortunately, our outdated bundled version of `@wordpress/components` (19.8.5) and the [deprecated `@types/wordpress__components`](https://www.npmjs.com/package/@types/wordpress__components/v/19.10.0) doesn't allow me to import `VStack` or `__experimentalVStack`, so I'd need to either override the type definitions manually or update the components or `@types` library versions, both out of scope for this UI change.

I'd also be concerned about using an early (3+ years old) version of an experimental component, when `Flex` seems to do the trick well enough.

Reply:
Got it. I strongly recommend updating `@wordpress/*` dependencies, since, as you also mention, they are 3+ years old. The package has been shipping its own types for a while, on top of adding several improvements and features.

Reply:
Yes 100% agree @ciampo, these problems are only going to get worse as time goes on. This is a know issue that we're prioritising in these P2 posts:

[Aligning WooPayments with the WordPress component package](paJDYF-gne-p2)
and 
[Spike: How WooPayments imports, bundles, and overrides styles from WP components](paJDYF-gpC-p2)